rs/consensus/src/consensus/purger.rs

```diff
@@ -324,7 +324,8 @@ impl Purger {
             .certified_height
             .min(self.state_manager.latest_state_height());
 
-        self.state_manager.remove_inmemory_states_below(height);
+        self.state_manager
+            .remove_inmemory_states_below(height, &BTreeSet::new());
```

Comment:
@eichhorl What do you think of the argument being a `&BTreeSet`?
Is it possible that Consensus is actively making a CUP from two summary blocks. Suppose the state manager creates checkpoint at height 500 and 1000. However, the manifest computation is extremely slow and root hash is not available for 500 even though we reach 1000. 
In this case, if we only protect the state at the height of `highest_summary_block`, we will still remove the in-memory state at 500.
My question is if the situation could happen and if so, should Consensus calls this API with all summary block heights that has not created a CUP yet.

Reply:
Good point! In practice this shouldn't happen, as the maximum gap between the notarized height and the highest "pending" CUP height is [set to a constant 70](https://sourcegraph.com/github.com/dfinity/ic/-/blob/rs/consensus/utils/src/lib.rs?L42-49). So in your example, we would effectively stall at height 570 until the CUP is created.

Regardless, it is likely a good idea to not rely on this constant and instead call the API with all summary heights for which there are no CUPs yet, as you suggest.